staking/apr/compute.go

```diff
@@ -55,152 +56,81 @@ func expectedRewardPerYear(
-	}
-
-	return *oneAgoHeader, *twoAgoHeader, nil
-}
-
 var (
 	zero = numeric.ZeroDec()
 )
 
 // ComputeForValidator ..
 func ComputeForValidator(
 	bc Reader,
-	now *big.Int,
-	state *state.DB,
+	block *types.Block,
 	validatorNow *staking.ValidatorWrapper,
-	blocksPerEpoch uint64,
 ) (*numeric.Dec, error) {
-	return &zero, nil
-	// twoEpochAgo, oneEpochAgo, zero :=
-	// 	new(big.Int).Sub(now, common.Big2),
-	// 	new(big.Int).Sub(now, common.Big1),
-	// 	numeric.ZeroDec()
-
-	// utils.Logger().Info().
-	// 	Uint64("now", now.Uint64()).
-	// 	Uint64("two-epoch-ago", twoEpochAgo.Uint64()).
-	// 	Uint64("one-epoch-ago", oneEpochAgo.Uint64()).
-	// 	Msg("apr - begin compute for validator ")
-
-	// twoSnapshotAgo, err := bc.ReadValidatorSnapshotAtEpoch(
-	// 	twoEpochAgo,
-	// 	validatorNow.Address,
-	// )
-
-	// if err != nil {
-	// 	return &zero, nil
-	// }
-
-	// oneSnapshotAgo, err := bc.ReadValidatorSnapshotAtEpoch(
-	// 	oneEpochAgo,
-	// 	validatorNow.Address,
-	// )
-
-	// if err != nil {
-	// 	return &zero, nil
-	// }
-
-	// blockNumAtTwoEpochAgo, blockNumAtOneEpochAgo :=
-	// 	shard.Schedule.EpochLastBlock(twoEpochAgo.Uint64()),
-	// 	shard.Schedule.EpochLastBlock(oneEpochAgo.Uint64())
-
-	// headerOneEpochAgo, headerTwoEpochAgo, err := pastTwoEpochHeaders(bc)
-
-	// // TODO Figure out why this is happening
-	// if headerOneEpochAgo == nil || headerTwoEpochAgo == nil || err != nil {
-	// 	utils.Logger().Debug().
-	// 		Msgf("apr compute headers epochs ago %+v %+v %+v %+v %+v %+v",
-	// 			twoEpochAgo, oneEpochAgo,
-	// 			blockNumAtTwoEpochAgo, blockNumAtOneEpochAgo,
-	// 			headerOneEpochAgo, headerTwoEpochAgo,
-	// 		)
-	// 	return &zero, nil
-	// }
-
-	// utils.Logger().Info().
-	// 	RawJSON("current-epoch-header", []byte(bc.CurrentHeader().String())).
-	// 	RawJSON("one-epoch-ago-header", []byte(headerOneEpochAgo.String())).
-	// 	RawJSON("two-epoch-ago-header", []byte(headerTwoEpochAgo.String())).
-	// 	Msg("headers used for apr computation")
-
-	// estimatedRewardPerYear, err := expectedRewardPerYear(
-	// 	headerOneEpochAgo, headerTwoEpochAgo,
-	// 	oneSnapshotAgo, twoSnapshotAgo,
-	// 	blocksPerEpoch,
-	// )
-
-	// if err != nil {
-	// 	return nil, err
-	// }
-
-	// if estimatedRewardPerYear.Cmp(common.Big0) == 0 {
-	// 	return &zero, nil
-	// }
-
-	// total := numeric.NewDecFromBigInt(validatorNow.TotalDelegation())
-	// if total.IsZero() {
-	// 	return nil, errors.New("zero total delegation will cause div by zero")
-	// }
-
-	// result := numeric.NewDecFromBigInt(estimatedRewardPerYear).Quo(
-	// 	total,
-	// )
-	// return &result, nil
+	oneEpochAgo, zero :=
+		new(big.Int).Sub(block.Epoch(), common.Big1),
+		numeric.ZeroDec()
+
+	utils.Logger().Info().
+		Uint64("now", block.Epoch().Uint64()).
+		Uint64("one-epoch-ago", oneEpochAgo.Uint64()).
+		Msg("apr - begin compute for validator ")
+
+	oneSnapshotAgo, err := bc.ReadValidatorSnapshotAtEpoch(
+		oneEpochAgo,
+		validatorNow.Address,
+	)
+
+	if err != nil {
+		return &zero, nil
```

Comment:
why discard the err?

Reply:
because this error should be a non-blocking error. We don't want to fail the block commitment because of this.

Reply:
yea, we shouldn't. Fixed.